packages/pass-style/src/copyBytes.js

```diff
@@ -0,0 +1,82 @@
+/// <reference types="ses"/>
+
+import { assertChecker } from './passStyle-helpers.js';
+
+/** @typedef {import('./types.js').CopyBytes} CopyBytes */
+
+const { Fail } = assert;
+const { setPrototypeOf } = Object;
+const { apply } = Reflect;
+
+/**
+ * @type {WeakSet<CopyBytes>}
+ */
+const genuineCopyBytes = new WeakSet();
+
+const slice = ArrayBuffer.prototype.slice;
+const sliceOf = (buffer, start, end) => apply(slice, buffer, [start, end]);
+
+/**
+ * A CopyBytes is much like an ArrayBuffer, but immutable.
+ * It cannot be used as an ArrayBuffer argument when a genuine ArrayBuffer is
+ * needed. But a `copyBytes.slice()` is a genuine ArrayBuffer, initially with
+ * a copy of the copyByte's data.
+ *
+ * On platforms that support freezing ArrayBuffer, like perhaps a future XS,
+ * (TODO) the intention is that `copyBytes` could hold on to a single frozen
+ * one and return it for every call to `arrayBuffer.slice`, rather than making
+ * a fresh copy each time.
+ *
+ * @param {ArrayBuffer} arrayBuffer
+ * @returns {CopyBytes}
+ */
+export const makeCopyBytes = arrayBuffer => {
+  try {
+    // Both validates and gets an exclusive copy.
+    // This `arrayBuffer` must not escape, to emulate immutability.
+    arrayBuffer = sliceOf(arrayBuffer);
+  } catch {
+    Fail`Expected genuine ArrayBuffer" ${arrayBuffer}`;
+  }
+  /** @type {CopyBytes} */
+  const copyBytes = {
+    // Can't say it this way because it confuses TypeScript
+    // __proto__: ArrayBuffer.prototype,
+    byteLength: arrayBuffer.byteLength,
+    slice(start, end) { return sliceOf(arrayBuffer, start, end); },
```

Comment:
a slice of a `CopyBytes` is a mutable `ArrayBuffer`? am i reading that right? that seems weird.

Reply:
Now, with https://github.com/endojs/endo/pull/2311 following @phoddie's suggestion at https://github.com/endojs/endo/pull/2309#issuecomment-2155513240 , this will be proposed as merely an ArrayBuffer.p.slice producing a normal mutable ArrayBuffer, independent of whether the original was mutable or immutable. Nothing could be less surprising. Resolving.